homeassistant/components/sensor/juicenet.py

```diff
@@ -0,0 +1,125 @@
+"""
+Support for monitoring juicenet/juicepoint/juicebox based EVSE sensors.
+
+For more details about this platform, please refer to the documentation at
+at https://home-assistant.io/components/sensor.juicenet/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.const import TEMP_CELSIUS
+from homeassistant.helpers.entity import Entity
+from homeassistant.components.juicenet import JuicenetDevice, DOMAIN
+
+DEPENDENCIES = ['juicenet']
+_LOGGER = logging.getLogger(__name__)
+
+SENSOR_TYPES = {
+    'status': ['Charging Status', None],
+    'temperature': ['Temperature', TEMP_CELSIUS],
+    'voltage': ['Voltage', 'V'],
+    'amps': ['Amps', 'A'],
+    'watts': ['Watts', 'W'],
+    'charge_time': ['Charge time', 's'],
+    'energy_added': ['Energy added', 'Wh']
+}
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Juicenet sensor."""
+    api = hass.data[DOMAIN]['api']
+
+    dev = []
+    for device in api.get_devices():
+        _id = device.id()
+        if _id not in hass.data[DOMAIN]['unique_ids']:
```

Comment:
You don't have to do this, Home Assistant does this automatically as long as you implement the property unique_id.